packages/blade/src/components/Menu/docs/CustomMenu.web.tsx

```diff
@@ -0,0 +1,143 @@
+import React from 'react';
+import styled from 'styled-components';
+import { MenuItem } from '../';
+import { Box } from '~components/Box';
+import {
+  BusinessBankingIcon,
+  CurrentAccountIcon,
+  RazorpayXIcon,
+  PaymentButtonIcon,
+  PaymentGatewayIcon,
+  PaymentLinksIcon,
+  PaymentPagesIcon,
+} from '~components/Icons';
+import type { IconComponent } from '~components/Icons';
+import { Text } from '~components/Typography';
+import { makeBorderSize, makeSize, makeSpace } from '~utils';
+import { getFocusRingStyles } from '~utils/getFocusRingStyles';
+
+const CustomMenuTrigger = styled.button<{ isTransparent?: boolean }>((props) => {
+  return {
+    all: 'unset',
+    padding: makeSpace(props.theme.spacing[4]),
+    color: props.theme.colors.surface.text.gray.normal,
+    backgroundColor: props.isTransparent
+      ? props.theme.colors.transparent
+      : props.theme.colors.interactive.background.gray.faded,
+    borderRadius: props.isTransparent ? '0px' : makeSize(props.theme.border.radius.medium),
+    borderBottomWidth: makeBorderSize(props.theme.border.width.thick),
+    borderBottomStyle: 'solid',
+    borderBottomColor: props.theme.colors.transparent,
+    cursor: 'pointer',
+    '&:hover, &.menu-open': {
```

Comment:
How will consumer know about this internal class called `.menu-open`?? 

Can we use aria attributes (eg: aria-expanded or something) to target the open state instead since that will be more "generic" and standard way to do it. 

Other libraries use the similar method, and some like radix also expose data attributes for the states for styling, eg `[data-open]`

Reply:
Good idea. Let me check

Reply:
Nice suggestion! Removed a lot of code